src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -1773,6 +1466,390 @@ subroutine interp_terrain(mesh, kdtree, geog_data_path)
 
  end subroutine interp_terrain
 
+!--------------------------------------------------------------------------------------------------
+! Categorical interpolations - Landuse and Soiltype
+!--------------------------------------------------------------------------------------------------
+
+ !***********************************************************************
+ !
+ !  routine categorical_interp_criteria
+ !
+ !> \brief   Categorical dataset interp criteria
+ !> \author  Miles Curry
+ !> \date    25 January 2020
+ !> \details
+ !> This routine can be used to determine if the tile that contains iCell needs
+ !> to be loaded and processed by init_atm_map_static_data for categorical datasets.
+ !
+ !-----------------------------------------------------------------------
+ function categorical_interp_criteria(iCell)
+
+     integer, intent(in) :: iCell
+     logical :: categorical_interp_criteria
+
+     categorical_interp_criteria = .false.
+
+     if (all(ncat(:,iCell) == 0)) then
+         categorical_interp_criteria = .true.
+     end if
+
+ end function categorical_interp_criteria
+
+ !***********************************************************************
+ !
+ !  routine categorical_interp_accumulation
+ !
+ !> \brief   Accumulate categorical dataset values
+ !> \author  Miles Curry
+ !> \date    25 January 2020
+ !> \details
+ !> This routine accumulates categorical values for the init_atm_map_static_data unified
+ !> function.
+ !
+ !-----------------------------------------------------------------------
+ subroutine categorical_interp_accumulation(iCell, cat)
+
+     integer, intent(in) :: iCell
+     integer (kind=I8KIND), dimension(:), intent(in) :: cat 
+     ! Use the module level category_min and category_max variables
+
+     if (cat(1) < category_min .or. cat(1) > category_max) then
+        !
+        ! Currently, the MODIS landuse dataset has zeros at the South Pole, and possibly other
+        ! places, so this return statement does not accumulate those values, which would be below ncat's
+        ! lower bound (aka, category_min). Other categorical datasets may want to use create a new
+        ! accumulation routine that produces an error when pixel(:) is above or below category_min/max.
+        !
+        return
+     else
+        ncat(cat(1), iCell) = ncat(cat(1), iCell) + 1
+     end if
```

Comment:
This if/else block still seems awkward to me. If we have an `else` condition for accumulating the category, then the `return` statement isn't necessary; on the other hand, if we keep the `return` statement, then the `else` condition isn't necessary. I would propose either of the following:
```fortran
     !
     ! Currently, the MODIS landuse dataset has zeros at the South Pole, and possibly other
     ! places, so we return early if given an invalid category
     !
     if (cat(1) < category_min .or. cat(1) > category_max) then
        return
     end if

     ncat(cat(1), iCell) = ncat(cat(1), iCell) + 1
```
or
```fortran
     !
     ! Currently, the MODIS landuse dataset has zeros at the South Pole, and possibly other
     ! places, so we need a check on the validity of the category to be accumulated
     !
     if (cat(1) >= category_min .and. cat(1) <= category_max) then
        ncat(cat(1), iCell) = ncat(cat(1), iCell) + 1
     end if
```

Between the two options, I think the second seems better to me.

Reply:
I agree as well. The if statement was a little wonky.